### PR TITLE
Add start date support for conversion filing and warnings check

### PR DESCRIPTION
**Pull request by lekshmimallika-aot**
*Issue #:* /bcgov/entity#13509

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

**Comment by lekshmimallika-aot** [comment]
Fix for two test failures in legal api is in  https://github.com/bcgov/lear/pull/1666

File: legal-api/src/legal_api/services/filings/validations/validation.py [hunk 1]
```diff
@@ -159,6 +160,9 @@ def validate(business: Business, filing_json: Dict) -> Error:  # pylint: disable
                 elif k == Filing.FILINGS['putBackOn'].get('name'):
                     err = put_back_on_validate(business, filing_json)
 
+                elif k == Filing.FILINGS['conversion'].get('name'):
+                    err = conversion_validate(business, filing_json)
+
```
Comment (argush3) [inline legal-api/src/legal_api/services/filings/validations/validation.py:165]: oh wow, we weren't even validating the conversion filing.  good catch

File: legal-api/tests/unit/services/filings/validations/test_conversion.py [hunk 2]
```diff
@@ -99,46 +115,83 @@ def test_invalid_nr_conversion(session):
...
+    'test_name, legal_type, start_date, filing, expected_msg',
+    [
+        ('sp_invalid_start_date', 'SP', '2016-01-01', copy.deepcopy(SP_CONVERSION),
+         'Start Date must be less than or equal to 2 years in the past and less than or equal to 90 days in the '+
+         'future from the registration date.'),
+        ('gp_invalid_start_date', 'GP', '2016-01-01', copy.deepcopy(GP_CONVERSION),
+         'Start Date must be less than or equal to 2 years in the past and less than or equal to 90 days in the future from the registration date.'),
+        ('sp_valid_start_date', 'SP', '2019-01-01', copy.deepcopy(SP_CONVERSION), None),
+        ('gp_valid_start_date', 'GP', '2019-01-01', copy.deepcopy(GP_CONVERSION), None),
+    ]
```
Comment (argush3) [inline legal-api/tests/unit/services/filings/validations/test_conversion.py:181]: can we add two test cases for the scenario where SP/GPs have a start date is more than 90 days after the registration date?

**Review by argush3 — COMMENTED** [review]
looks good, just the comment about adding two more parametrized tests

**Review by vysakh-menon-aot — APPROVED** [review]

File: legal-api/tests/unit/services/filings/validations/test_conversion.py (hunk 2 above)
Comment (lekshmimallika-aot) [reply legal-api/tests/unit/services/filings/validations/test_conversion.py]: Done

**Review by argush3 — APPROVED** [review]

**Pull request merged**